service_taxi/src/main/kotlin/org/thechance/service_taxi/api/models/TaxiDto.kt

```diff
@@ -1,29 +1,42 @@
 package org.thechance.service_taxi.api.models
 
 import kotlinx.serialization.Serializable
+import org.bson.types.ObjectId
 import org.thechance.service_taxi.domain.entity.Taxi
 
 @Serializable
 data class TaxiDto(
-    val id: String,
-    val plateNumber: String,
-    val color: String,
-    val type: String,
-    val driverId: String,
-    val isAvailable: Boolean = true,
-    val capacity: Int,
+    val id: String? = null,
+    val plateNumber: String? = null,
+    val color: String? = null,
+    val type: String? = null,
+    val driverId: String? = null,
+    val isAvailable: Boolean? = null,
+    val capacity: Int? = null,
 )
 
 // region mappers
+fun TaxiDto.toTaxi(): Taxi {
+    return Taxi(
+        id = if (id.isNullOrBlank()) ObjectId().toHexString() else ObjectId(id).toHexString(),
```

Comment:
`id = if (id.isNullOrBlank()) ObjectId().toHexString() else ObjectId(id).toHexString(),
`I think this is not here ,maybe

Reply:
Why ?
It needed for mapping correctly